akka-actor-typed-tests/src/test/scala/docs/akka/typed/StashDocSpec.scala

```diff
@@ -4,90 +4,107 @@
+  import scala.util.Failure
+  import scala.util.Success
+
+  import akka.Done
+  import akka.actor.typed.ActorRef
+  import akka.actor.typed.Behavior
+  import akka.actor.typed.scaladsl.ActorContext
+  import akka.actor.typed.scaladsl.Behaviors
+  import akka.actor.typed.scaladsl.StashBuffer
+
   trait DB {
     def save(id: String, value: String): Future[Done]
     def load(id: String): Future[String]
   }
 
   object DataAccess {
-    trait Command
+    sealed trait Command
     final case class Save(value: String, replyTo: ActorRef[Done]) extends Command
     final case class Get(replyTo: ActorRef[String]) extends Command
     private final case class InitialState(value: String) extends Command
     private final case object SaveSuccess extends Command
     private final case class DBError(cause: Throwable) extends Command
 
-    def behavior(id: String, db: DB): Behavior[Command] =
+    def apply(id: String, db: DB): Behavior[Command] = {
       Behaviors.withStash(100) { buffer =>
         Behaviors.setup[Command] { context =>
-          def init(): Behavior[Command] =
-            Behaviors.receiveMessage[Command] {
-              case InitialState(value) =>
-                // now we are ready to handle stashed messages if any
-                buffer.unstashAll(active(value))
-              case DBError(cause) =>
-                throw cause
-              case other =>
-                // stash all other messages for later processing
-                buffer.stash(other)
-                Behaviors.same
-            }
-
-          def active(state: String): Behavior[Command] =
-            Behaviors.receive { (context, message) =>
-              message match {
-                case Get(replyTo) =>
-                  replyTo ! state
-                  Behaviors.same
-                case Save(value, replyTo) =>
-                  context.pipeToSelf(db.save(id, value)) {
-                    case Success(_)     => SaveSuccess
-                    case Failure(cause) => DBError(cause)
-                  }
-                  saving(value, replyTo)
-              }
-            }
-
-          def saving(state: String, replyTo: ActorRef[Done]): Behavior[Command] =
-            Behaviors.receiveMessage[Command] {
-              case SaveSuccess =>
-                replyTo ! Done
-                buffer.unstashAll(active(state))
-              case DBError(cause) =>
-                throw cause
-              case other =>
-                buffer.stash(other)
-                Behaviors.same
-            }
-
-          context.pipeToSelf(db.load(id)) {
-            case Success(value) => InitialState(value)
+          new DataAccess(context, buffer, id, db).init()
+        }
+      }
+    }
+  }
+
+  class DataAccess(
+      context: ActorContext[DataAccess.Command],
+      buffer: StashBuffer[DataAccess.Command],
+      id: String,
+      db: DB) {
+    import DataAccess._
+
+    private def init(): Behavior[Command] = {
```

Comment:
Strikes me more as `start()`, as start functions are typically Unit return types, where initialize functions take initialization params.

Reply:
ok, changed to `start`